src/broadcast/broadcastmanager.cpp

```diff
@@ -6,16 +6,20 @@
 #include "engine/sidechain/enginesidechain.h"
 #include "soundio/soundmanager.h"
 
-BroadcastManager::BroadcastManager(UserSettingsPointer pConfig,
+BroadcastManager::BroadcastManager(SettingsManager* pSettingsManager,
                                    SoundManager* pSoundManager)
-        : m_pConfig(pConfig) {
+        : m_pConfig(pSettingsManager->settings()) {
     QSharedPointer<EngineNetworkStream> pNetworkStream =
             pSoundManager->getNetworkStream();
+
     if (!pNetworkStream.isNull()) {
         m_pBroadcast = QSharedPointer<EngineBroadcast>(
```

Comment:
Can we simplify the code somehow? It is unclear for me if we have a Shared pointer dead lock here or not. 
Is it possible to turn some of these instances into a unique pointer, to clean up the ownership here?